lib/ConnectionInfo.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace ActiveRecord;
+
+class ConnectionInfo {
+
+	public $protocol = null;
+	public $host = null;
+	public $port = null;
+	public $database = null;
+
+	public $username = null;
+	public $password = null;
+
+	public function __construct($input = array()){
+		foreach($input as $prop => $value){
+			$this->{$prop} = $value;
```

Comment:
I would check [if the property exists](http://php.net/manual/en/function.property-exists.php) before assigning it.